controllers/imagerepository_notifications.go

```diff
@@ -0,0 +1,189 @@
+	log := ctrllog.FromContext(ctx).WithName("HandleNotifications")
+
+	if imageRepository.Status.Notifications == nil {
+		// No status notifications to check
+		return nil
+	}
+	for index := 0; index < len(imageRepository.Status.Notifications); index++ {
+		statusNotification := &imageRepository.Status.Notifications[index] // This way we can update the UUID if notification gets updated
+		existsInSpec := false
+		for _, notification := range imageRepository.Spec.Notifications {
+			if notification.Title == statusNotification.Title {
+				existsInSpec = true
+				quayNotification, err := r.notificationExistsInQuayByUUID(statusNotification.UUID, imageRepository)
+				if err != nil {
+					return r.handleError(ctx, imageRepository, err, "Couldn't retrieve all Quay notifications")
+				}
+				if isNotificationChanged(notification, quayNotification) {
+					// Updated item in Spec.Notifications: update notification in Quay
+					log.Info("Updating notification in Quay", "Title", notification.Title, "Event", notification.Event, "Method", notification.Method, "UUID", quayNotification.UUID)
+					updatedNotification, err := r.QuayClient.UpdateNotification(
+						r.QuayOrganization,
+						imageRepository.Spec.Image.Name,
+						statusNotification.UUID,
+						quay.Notification{
+							Title:  notification.Title,
+							Event:  string(notification.Event),
+							Method: string(notification.Method),
+							Config: quay.NotificationConfig{
+								Url: notification.Config.Url,
+							},
+							EventConfig: quay.NotificationEventConfig{},
+						})
+					if err != nil {
+						log.Error(err, "failed to update notification", "Title", statusNotification.Title, "UUID", statusNotification.UUID)
+						return r.handleError(ctx, imageRepository, err, "Error while updating notification ("+notification.Title+")")
+					}
+					statusNotification.UUID = updatedNotification.UUID
+					statusNotification.Title = updatedNotification.Title
+					break
+				}
+			} else {
+				exists, err := r.notificationExistsInQuayByTitle(notification.Title, imageRepository)
+				if err != nil {
+					return r.handleError(ctx, imageRepository, err, "Couldn't retrieve all Quay notifications")
+				}
+				if !exists {
+					// New item in Spec.Notifications: add notification to Quay and Status.Notifications
+					log.Info("Adding new notification to Quay", "Title", notification.Title, "Event", notification.Event, "Method", notification.Method)
+					notificationStatus, err := r.AddNotification(notification, imageRepository)
+					if err != nil {
+						log.Error(err, "failed to add notification", "Title", notification.Title, "Event", notification.Event, "Method", notification.Method)
+						return r.handleError(ctx, imageRepository, err, "Error while adding a notification ("+notification.Title+") to Quay")
+					}
+					imageRepository.Status.Notifications = append(imageRepository.Status.Notifications, notificationStatus)
```

Comment:
you should also check here IF the notitification isn't already in the status and add it only if it is missing, to avoid duplicity in status

Reply:
Discussed via Slack. Concluded to update PR for safety reasons.